### PR TITLE
Handle `zorder` between overlayed cartesian subplots

### PR DESCRIPTION
**Pull request by archmoj**
Supersedes #6943.
Fixes #6943.

Changes to make `zorder` work between overlayed cartesian subplots!

### CODE REVIEW

**Review by Farkites — APPROVED** [review]
Works like a charm!!! I have tested the sample code provided by the sponsors and it behaves as expected. Other cases also work correctly. Good job!! 💃 Thanks @archmoj!

**Comment by gvwilson** [comment]
woo hoo!

File: test/jasmine/tests/splom_test.js
[the remark this answers is not in the document]
Comment (archmoj) [reply test/jasmine/tests/splom_test.js:1]: Yes. Please note that the first commits are refactors required to place the main subplot inside the same container as overlayed subplots. This change was necessary to let use add the traces depending on their `zorder` value. The changes in tests are showing this.

[the remark this answers is not in the document]
Comment (Farkites) [reply test/jasmine/tests/splom_test.js:1]: The changes in the other tests made sense to me but I was unsure about this one. Thanks for clarifying!

**Comment by emilykl** [comment]
Awesome PR @archmoj 🏆 🌟 

Am I understanding the code correctly, that the approach here is simply to create separate subplots for each individual `zorder` value, and plot them from lowest to highest? Makes sense to me.

**Comment by archmoj** [comment]
> Awesome PR @archmoj 🏆 🌟
> 
> Am I understanding the code correctly, that the approach here is simply to create separate subplots for each individual `zorder` value, and plot them from lowest to highest? Makes sense to me.

@emilykl yes that's correct.
I'm wondering now that with this in place is it possible to simplify some of the logic added in #6918?

**Pull request merged**